packages/dialtone-vue2/components/rich_text_editor/extensions/emoji/emoji.js

```diff
@@ -1,29 +1,29 @@
 import { mergeAttributes, Node, nodeInputRule, nodePasteRule } from '@tiptap/core';
 import { VueNodeViewRenderer } from '@tiptap/vue-2';
 import EmojiComponent from './EmojiComponent.vue';
-import { codeToEmojiData } from '@/common/emoji';
+import { codeToEmojiData, emojiShortCodeRegex } from '@/common/emoji';
 import { PluginKey } from '@tiptap/pm/state';
 
 import Suggestion from '@tiptap/suggestion';
 import suggestionOptions from './suggestion';
 
 export const EmojiPluginKey = new PluginKey('emoji');
 
-const inputShortCodeRegex = /:\w+:$/;
-export const emojiShortCodeRegex = /:\w+:/g;
+const inputShortCodeRegex = /(^| )(:\w+:)$/;
```

Comment:
Is it necessary to have this? Maybe we can just make our const a regex string `"(^| )(:\w+:)"` and then append a $ to it here? then use `new RegExp` to turn the string into a regex? Not that big of deal though.

Reply:
> Is it necessary to have this? Maybe we can just make our const a regex string `"(^| )(:\w+:)"` and then append a $ to it here? then use `new RegExp` to turn the string into a regex? Not that big of deal though.

Yes, most places use `emojiShortCodeRegex` which already includes a global flag, I know is not ideal but seems better that having a string and having to call `new RegExp()` everywhere adding the `g` flag.